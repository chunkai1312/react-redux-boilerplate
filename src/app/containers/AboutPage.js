import React, { Component, PropTypes } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import * as actions from '../actions/application'

class AboutPage extends Component {

  componentWillMount () {
    this.props.actions.setPageTitle('About')
  }

  render () {
    return (
      <div>
        About
      </div>
    )
  }
}

AboutPage.propTypes = {
  application: PropTypes.object.isRequired,
  actions: PropTypes.object.isRequired
}

const mapStateToProps = (state) => ({
  application: state.application
})

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators(actions, dispatch)
})

export default connect(mapStateToProps, mapDispatchToProps)(AboutPage)
