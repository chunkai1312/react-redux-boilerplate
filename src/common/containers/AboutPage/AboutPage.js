import React, { Component, PropTypes } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import Helmet from 'react-helmet'
import { MainContent } from '../../components/AppLayout'
import * as actions from '../../actions'

class AboutPage extends Component {
  static propTypes = {
    appLayout: PropTypes.object.isRequired,
    actions: PropTypes.object.isRequired
  }

  componentDidMount () {
    this.props.actions.toggleBackButton()
  }

  componentWillUnmount () {
    this.props.actions.toggleBackButton()
  }

  render () {
    return (
      <MainContent>
        <Helmet title="About" />
        About
      </MainContent>
    )
  }
}

const mapStateToProps = (state) => ({
  appLayout: state.appLayout
})

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators(actions, dispatch)
})

export default connect(mapStateToProps, mapDispatchToProps)(AboutPage)
