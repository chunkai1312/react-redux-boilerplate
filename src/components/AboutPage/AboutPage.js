import React from 'react'
import Helmet from 'react-helmet'

const AboutPage = props => {
  return (
    <div>
      <Helmet title="About" />About
    </div>
  )
}

export default AboutPage
