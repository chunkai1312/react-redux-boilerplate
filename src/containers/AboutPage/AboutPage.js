import React, { Component, PropTypes } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import * as actions from '../../actions/layout'

class AboutPage extends Component {
  static propTypes = {
    layout: PropTypes.object.isRequired,
    actions: PropTypes.object.isRequired
  }

  componentWillMount () {
    this.props.actions.setPageTitle('About')
  }

  render () {
    return (
      <div style={{ padding: '2.4rem' }}>
        About
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  layout: state.layout
})

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators(actions, dispatch)
})

export default connect(mapStateToProps, mapDispatchToProps)(AboutPage)
