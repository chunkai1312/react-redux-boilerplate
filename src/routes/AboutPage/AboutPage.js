import React, { Component, PropTypes } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { MainContent } from 'components/AppLayout'
import * as actions from '../../actions'

class AboutPage extends Component {
  static propTypes = {
    appLayout: PropTypes.object.isRequired,
    actions: PropTypes.object.isRequired
  }

  componentWillMount () {
    this.props.actions.toggleBackButton()
  }

  componentWillUnmount () {
    this.props.actions.toggleBackButton()
  }

  render () {
    return (
      <MainContent>
        About
      </MainContent>
    )
  }
}

const mapStateToProps = (state) => ({
  appLayout: state.appLayout
})

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators(actions, dispatch)
})

export default connect(mapStateToProps, mapDispatchToProps)(AboutPage)
