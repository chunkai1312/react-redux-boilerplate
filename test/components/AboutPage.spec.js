import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Helmet from 'react-helmet'
import AboutPage from '../../src/components/AboutPage'

function setup () {
  const wrapper = shallow(<AboutPage />)
  return { wrapper }
}

describe('<AboutPage />', () => {
  it('should render a <Helmet /> component', () => {
    const { wrapper } = setup()
    expect(wrapper.find(Helmet)).toHaveLength(1)
    expect(wrapper.find(Helmet).node.props.title).toBe('About')
  })

  it('should render correctly', () => {
    const { wrapper } = setup()
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
