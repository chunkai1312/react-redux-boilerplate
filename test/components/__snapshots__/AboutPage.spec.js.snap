// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutPage /> should render correctly 1`] = `
<div>
  <HelmetWrapper
    encodeSpecialCharacters={true}
    title="About"
  />
  About
</div>
`;
